modules/react/tabs/lib/TabsList.tsx

```diff
@@ -44,12 +44,12 @@ export const useTabsList = composeHooks(
 );
 
 const StyledStack = styled(Stack)<StyledType>({
+  maskImage: `linear-gradient(to right, white 95%, transparent)`,
   '::after': {
     content: '""',
     position: 'sticky',
     height: 52,
     minWidth: 30,
-    background: `linear-gradient(to right,rgba(255,255,255,0),white);`,
```

Comment:
Thanks for taking this on @anishatulai! After reviewing your PR, this has turned out to be a more complicated issue than I anticipated 🤔 

I noticed `mask-image` doesn't work when applied to the `::after` pseudo-element: the `-webkit-mask-image/mask-image` style is applied, but it's not producing the desired visual fade effect. I'm guessing this is why you chose to apply `maskImage` to the StyledStack instead.

Unfortunately, this produces an undesired fade effect for users aren't using a touch device. This is a screenshot of the [Basic Tabs story](https://5d854c26ba934e0020f5e98a-aoijkqciui.chromatic.com/?path=/story/components-containers-tabs-react--basic) in the current preview build:

<img width="1068" alt="undesired-fade" src="https://user-images.githubusercontent.com/146020/182500871-4b401ba3-e916-46d3-ab47-9c3a94543ca8.png">

Note how the thin gray bottom border of the tab list fades out at the very right (it's very subtle, but it's there). This is what's triggering the regression in the Chromatic test.

---

We could potentially address this by only applying `maskImage` to the StyledStack if the `modality` is set to `touch`. The subtle fade out on the right edge of the tab list border would still be present on touch devices (it wasn't before), but that may be acceptable. Here are before and after screenshots from my phone:

**Before**

![phone-before](https://user-images.githubusercontent.com/146020/182505152-55eda1d9-70f2-45dc-a297-69a36a063981.jpg)

**After** (only applying `maskImage` if the `modality` is set to `touch`)

![phone-after](https://user-images.githubusercontent.com/146020/182505178-90ad4d64-5456-4b9a-a7e1-44d04a4b5f9e.jpg)

---

Finally, we would also need to flip the direction of the `maskImage` gradient for RTL. Note the incorrect placement of the fade effect in the [RTL story](https://5d854c26ba934e0020f5e98a-aoijkqciui.chromatic.com/?path=/story/components-containers-tabs-react--right-to-left) in the current preview build (this is also causing a visual regression in the Chromatic test):

<img width="441" alt="fade-rtl" src="https://user-images.githubusercontent.com/146020/182503711-ed04499e-b918-4a2b-88cb-02a5c4febc9b.png">

Reply:
I actually first tried adjusting the linear gradient background colors to match whatever is set on the element but that didn't look right. The mask-image, as you have seen, only works on the StyledStack itself. I can look into it more to see if there is any way to avoid the gray line fading as well. And I completely forgot about the RTL, good catch! Thanks for the feedback @jamesfan ! I will make the adjustments.

Reply:
The fade doesn't bother me, but it may help if the fade is only apparent when the scrolling overflow is active (mobile) and when there is room to scroll. If you implement the latter, the former isn't necessary.

Reply:
Updated to only show fade effect on touch devices. Fixed RTL support.